src/typedefs.jl

```diff
@@ -82,8 +82,8 @@ the order of the pool's [`levels`](@ref DataAPI.levels) is used rather than the
 ordering of values of type `T`.
 """
 struct CategoricalValue{T <: SupportedTypes, R <: Integer}
-    level::R
     pool::CategoricalPool{T, R, CategoricalValue{T, R}}
+    ref::R
```

Comment:
smart - I was thinking how you have changed it - and it is just super simple :).

Reply:
Actually I wanted to change their order for a long time, but there's never been a strong enough reason to do it. :-)